### DISCUSSION TITLE
Forcing math.equation into sans mode will double the equation number

### DISCUSSION THREAD

**Discussion by cmarcotte** [typst/typst#2036] [Q&A] [answered]
Using the snippet shown [here](https://github.com/typst/typst/discussions/1497) will result in the equation numbering being doubled in the block compared to the reference. Example follows:
```typ
#show math.equation: it => { 
  if it.body.func() == math.sans[].func(){ 
    it 
  } else { 
    math.equation(block: it.block, math.sans(it)) 
  } 
} 
#set math.equation(numbering: "1.") 
$ g(s) = integral_(0)^(oo) f(t) e^(-s t) dif t $ <eq:1> 
This is a reference to @eq:1.

$ f(t) = integral_(0)^(oo) g(s) e^(s t) dif s $ <eq:2>
This is a reference to @eq:2.
```

Which displays equations 2 & 4, but references equations 1 & 2. Is this a known issue? Is there a better way to do this?

**Comment by freundTech** [comment, marked as answer]
The problem is that you are creating a second `math.equation` element inside your show rule, creating two `math.equation` elements for every one in your source code.

You can work around this by manually decrementing the equation counter before creating the second `math.equation` in the show rule:
```typst
#show math.equation: it => {
  if it.body.func() == math.sans[].func(){ 
    it 
  } else { 
    counter(math.equation).update(x => x - 1)
    math.equation(block: it.block, math.sans(it)) 
  } 
}
```
